var/spack/repos/builtin/packages/rust/package.py

```diff
@@ -4,59 +4,426 @@
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
 from spack import *
+from six import iteritems
 
 
 class Rust(Package):
-    """The rust programming language toolchain"""
+    """The Rust programming language toolchain
 
-    homepage = "http://www.rust-lang.org"
-    git      = "https://github.com/rust-lang/rust.git"
+    This package can bootstrap any version of the Rust compiler since Rust
+    1.23. It does this by downloading the platform-appropriate binary
+    distribution of the desired version of the rust compiler, and then building
+    that compiler from source.
+    """
 
-    version('develop', branch='master')
-    version('1.41.0', tag='1.41.0')
-    version('1.34.0', tag='1.34.0')
-    version('1.32.0', tag='1.32.0')
-    version('1.31.1', tag='1.31.1')
-    version('1.31.0', tag='1.31.0')  # "Rust 2018" edition
-    version('1.30.1', tag='1.30.1')
+    homepage = "https://www.rust-lang.org"
+    url = "https://static.rust-lang.org/dist/rustc-1.42.0-src.tar.gz"
+
+    maintainers = ["AndrewGaspar"]
+
+    phases = ['configure', 'build', 'install']
 
     extendable = True
 
-    # Rust
-    depends_on("llvm")
-    depends_on("curl")
-    depends_on("git")
-    depends_on("cmake")
-    depends_on("binutils")
-    depends_on("python@:2.8")
+    variant(
+        'rustfmt',
+        default=True,
+        description='Formatting tool for Rust code'
+    )
+
+    variant(
+        'analysis',
+        default=True,
+        description='Outputs code analysis that can be consumed by other tools'
+    )
+
+    variant(
+        'clippy',
+        default=True,
+        description='Linting tool for Rust'
+    )
+
+    variant(
+        'rls',
+        default=False,
+        description='The Rust Language Server can be used for IDE integration'
+    )
+
+    variant(
+        'src',
+        default=True,
+        description='Install Rust source files'
+    )
+
+    depends_on('cmake', type='build')
+    # We don't use binutils on Mac - we pick up ar either from the system or
+    # compiler
+    depends_on('binutils', type='build', when='platform=linux')
+    depends_on('binutils', type='build', when='platform=cray')
```

Comment:
This dependency is generally discouraged: https://spack.readthedocs.io/en/latest/getting_started.html#binutils